locksmith/src/controllers/hookController.ts

```diff
@@ -15,178 +26,141 @@ const Hub = z.object({
   secret: z.string().optional(),
 })
 
-const EXPIRATION_SECONDS_LIMIT = 86400 * 90
-
-export function getExpiration(leaseSeconds?: number) {
-  const limit = leaseSeconds ?? 864000
-  if (limit > EXPIRATION_SECONDS_LIMIT) {
-    throw new Error("Lease seconds can't be greater than 90 days")
+interface HookControllerOptions {
+  leaseSeconds: {
+    limit: number
+    default: number
   }
-  return new Date(Date.now() + limit * 1000)
 }
+export class HookController {
+  // eslint-disable-next-line
+  constructor(public options: HookControllerOptions) {}
 
-export async function subscribe(
-  hub: z.infer<typeof Hub>,
-  params: Record<string, string>
-) {
-  try {
-    const hook = await Hook.findOne({
-      where: {
-        topic: hub.topic,
-        callback: hub.callback,
-      },
-    })
-
-    if (!hook) {
-      await validHubIntent(hub)
-
-      const expiration = getExpiration(hub.lease_seconds)
-
-      const createdHook = await Hook.create({
-        expiration,
-        topic: hub.topic,
-        callback: hub.callback,
-        mode: hub.mode,
-        secret: hub.secret,
-        network: params.network,
-        lock: params.lock,
-      })
-
-      return createdHook
-    }
-
-    if (hook.mode !== hub.mode) {
-      hook.mode = hub.mode
+  async handle(request: SubscribeRequest, response: Response) {
+    const network = this.getNetwork(request.params.network)
+    if (!network) {
+      return response.status(404).send('Unsupported Network')
     }
-
-    if (hub.lease_seconds) {
-      hook.expiration = getExpiration(hub.lease_seconds)
+    const result = await Hub.safeParseAsync(request.body.hub)
+    if (!result.success) {
+      return response.status(400).send({
+        hub: {
+          mode: request.body.hub.mode,
+          topic: request.body.hub.topic,
+          reason: result.error.message,
+        },
+      })
+    } else {
+      // Send the accepted request to the subscriber and then validate the intent of the subscriber as well as persist the subscription.
+      response.status(202).send('Accepted')
+      try {
+        await this.verifySubscriber(request)
+        await this.updateHook(request)
+        return
+      } catch (error) {
+        logger.error(error.message)
+        // eslint-disable-next-line
+        return
+      }
     }
+  }
 
-    if (hub.secret) {
-      hook.secret = hub.secret
+  // Create an expiration date based on lease_seconds
+  getExpiration(leaseSeconds?: number) {
+    const limit = leaseSeconds ?? this.options.leaseSeconds.default
+    if (limit > this.options.leaseSeconds.limit) {
```

Comment:
We should not throw but replace the lease with the max value

Reply:
Okay will do.